samples/subsys/zbus/benchmark/src/benchmark.c

```diff
@@ -11,8 +11,19 @@
 #include <zephyr/logging/log.h>
 #include <zephyr/sys/util_macro.h>
 #include <zephyr/zbus/zbus.h>
+
+#if defined(CONFIG_ARCH_POSIX)
+#include "native_rtc.h"
+#define GET_ARCH_TIME_NS() (native_rtc_gettime_us(RTC_CLOCK_PSEUDOHOSTREALTIME) * NSEC_PER_USEC)
+#else
+#define GET_ARCH_TIME_NS() (k_cyc_to_ns_near32(sys_clock_cycle_get_32()))
```

Comment:
I was seeing issues on the ARC QEMU platform where the tick count runs at 100Hz; does this API provide a higher resolution value for that case?

Reply:
Seems so
```
-- west build: running target run_qemu
[0/1] To exit from QEMU enter: 'CTRL+a, x'[QEMU] CPU: arcem
*** Booting Zephyr OS build zephyr-v3.2.0-3158-g7bb6557ba710 ***
I: Benchmark 1 to 8: Dynamic memory, SYNC transmission and message size 256
I: Bytes sent = 262144, received = 262144
I: Average data rate: 2.11MB/s
I: Duration: 118061.400us

@118061400
```
```
[0/1] To exit from QEMU enter: 'CTRL+a, x'[QEMU] CPU: archs
*** Booting Zephyr OS build zephyr-v3.2.0-3158-g7bb6557ba710 ***
I: Benchmark 1 to 8: Dynamic memory, SYNC transmission and message size 256
I: Bytes sent = 262144, received = 262144
I: Average data rate: 2.12MB/s
I: Duration: 117709.400us

@117709400
```
```
[0/1] To exit from QEMU enter: 'CTRL+a, x'[QEMU] CPU: hs6x
*** Booting Zephyr OS build zephyr-v3.2.0-3158-g7bb6557ba710 ***
I: Benchmark 1 to 8: Dynamic memory, SYNC transmission and message size 256
I: Bytes sent = 262144, received = 262144
I: Average data rate: 3.12MB/s
I: Duration: 80011.600us

@80011600
```